index.js

```diff
@@ -241,7 +241,10 @@ function fastifyView (fastify, opts, next) {
 
     try {
       const html = render(page, data)
-      this.header('Content-Type', 'text/html; charset=' + charset).send(html)
+      if (!this.getHeader('content-type')) {
+        this.header('Content-Type', 'text/html; charset=' + charset)
+      }
```

Comment:
I don’t understand why this change would fix this, would you mind adding a comment or explaining it in the PR?

Reply:
this.header('Content-Type', 'text/html; charset=' + charset)

This statement returns undefined
undefined.send(html) 
This execution will report error

Reply:
Are you sure? `.header` should only ever return the `Reply` instance — https://github.com/fastify/fastify/blob/639298b38ae2e16ee3c6e951614145a892964412/lib/reply.js#L176

Reply:
![image](https://user-images.githubusercontent.com/3773790/63271515-eace0d00-c2cc-11e9-9f2a-813ef41eabbb.png)
fastify.view will take this logic

Reply:
A better fix is to have `header` return the instance in that mock.

Reply:
That condition is written also for others template's engine, so it would be better to use always the same logic

Reply:
so the final solution is？Hope to solve it as soon as possible，thank you

Reply:
so the final solution is？I hope to use this function in a few days, thanks!

Reply:
My opinion is that these implementations are wrong:

https://github.com/fastify/point-of-view/blob/b7345f0601ec0e789cae535cf96bc14ac42ac04d/index.js#L52-L53

They are replicating the core `Reply` methods and should behave accordingly. In fact, I'm not sure why the `view` decoration is on the server object. Views should be rendered in the context of a request, and in particular the context of sending a reply. So I would think the `view` decoration should be `fastify.decorateReply('view', function () {})`. Which would mean the following would be an outline of a solution:

```js
fastify.decorateReply('view', function () {
  // other details omitted
  renderer.apply({getHeader, header}, args)

  function getHeader (key) {
    return this.getHeader(key)
  }

  function header (key, val) {
    return this.header(key, val)
  }
})
```

Reply:
I guess i'd try to create a corresponding PR since this issue is blocking for us and i'd like to fix that without monkeypatching. I'm sure it would take me a few days to make it.

Reply:
@mcollina do you agree with my comment?

Reply:
In addition to @jsumners suggestion i'd say `fastify`as well as the `reply` object should be decorated. It might be necessary to customize the reply further or use the views to f.e. generate email-content as well.

Without any offences - it seems like the code-structure could be optimized (there are a lot of redundancies) in general to make future maintenance easier and to clean up some individual behaviors of the template engines.